### PR TITLE
Wrap jQuery plugin in a self-executing anonymous function.

### PR DESCRIPTION
**Pull request by vhinic**
By wrapping a jQuery plugin in a self-executing anonymous function, and passing in jQuery as an argument, the plugin author can safely use the $ alias within the plugin code and still know that the code will work whether or not jQuery.noConflict() is called by someone who uses the plugin.  This also prevents any variables that are introduced during the plugin's creation from leaking out into the global scope.

This also allow us to use this plugin in Drupal directly.

### CODE REVIEW

**Comment by afeld** [comment]
I am so sorry for not responding sooner. Looks like this pull request now includes a bit more than was originally intended...? For the new feature, could you tell me a little more about the use case?

File: test/toc-test.js [hunk 1]
```diff
@@ -73,6 +73,55 @@ describe('Toc', function() {
...
+            '<h3></h3>' +
+            '<h4></h4>' +
+            '<h5></h5>' +
+            '<h6></h6>' +
+            '<h6></h6>' +
+          '</div>'
+        );
+        var level = Toc.helpers.getTopLevel($scope, 1, true);
+        expect(level).to.eql(1);
+      });
```
Comment (afeld) [inline test/toc-test.js:91]: In this example, the table of contents would only show the `<h1>`, with the one `<h2>` beneath it...that's the desired behavior?

**Comment by vhinic** [comment]
Hi @afeld things for replying. Yes there are two more commits that are somewhat related.

First one is to include headers even if there is only one sibling. I have a client that is using your library in the Drupal and as they add content, they want to see side menu forming. So, we have situation where there will be just one H1 and one H2 or any parent-child situation with only one sibling.

Another commit is for situation (startLevel) where desired outcome is to ignore top level H and just start from some point onward. In the case of my client, we do not show H1 and just start with H2s down.

And thanks for the great library!

**Comment by afeld** [comment]
> include headers even if there is only one sibling

Would this (existing option) work?

``` javascript
$scope: $('h1,h2')
```

> ignore top level H and just start from some point onward

Similarly,

``` javascript
$scope: $('h2,h3,h4,h5,h6')
```

?

**Comment by afeld** [comment]
Closing as stale.

**Pull request closed without merging**